openlibrary/core/models.py

```diff
@@ -626,17 +626,24 @@ def resolve_redirect_chain(
                 Observations.get_observations_for_work(olid)
             )
 
-            # track updates
-            r['updates']['readinglog'] = Bookshelves.update_work_id(
-                olid, new_olid, _test=test
-            )
-            r['updates']['ratings'] = Ratings.update_work_id(olid, new_olid, _test=test)
-            r['updates']['booknotes'] = Booknotes.update_work_id(
-                olid, new_olid, _test=test
-            )
-            r['updates']['observations'] = Observations.update_work_id(
-                olid, new_olid, _test=test
-            )
+            if new_olid != olid:
+                # track updates
+                r['updates']['readinglog'] = Bookshelves.update_work_id(
+                    olid, new_olid, _test=test
+                )
+                r['updates']['ratings'] = Ratings.update_work_id(olid, new_olid, _test=test)
+                r['updates']['booknotes'] = Booknotes.update_work_id(
+                    olid, new_olid, _test=test
+                )
+                r['updates']['observations'] = Observations.update_work_id(
+                    olid, new_olid, _test=test
+                )
+                summary['modified'] = summary['modified'] or any(
+                    any(r['updates'][group].values()) for group in [
+                        'readinglog', 'ratings', 'booknotes', 'observations'
+                    ]
+                )
```

Comment:
This is a bit of a doozie.

It's effectively looking through the `summary`'s `updates` dict (which shows which tables have changed) and is checking to see if **any** of the values of **any** of these tables are non-zero.

This logic is being run on work in the chain (except the final/terminal resolved_work)

i.e. for every non-terminal work, keep track of whether any update has been made to any table.